src/components/GenreSelector/GenreSelector.tsx

```diff
@@ -1,21 +1,38 @@
 import React from "react";
+import { I18yGenres } from "../../core/constants/I18yConstants";
 
 import styles from "./GenreSelector.module.scss";
 
 interface GenreSelectorProps {
-  genres: string[];
+  genres: I18yGenres;
+  state: string;
+  handler: (g: string) => void;
 }
 
-const genreMapper = (genre: string, index: number): JSX.Element => {
-  return (
-    <div className={index === 0 ? styles.selected : ""} key={genre}>
-      <span>{genre}</span>
-    </div>
-  );
+const genreMapper = ({
+  genres,
+  state,
+  handler,
+}: GenreSelectorProps): JSX.Element[] => {
+  const mapped = Object.entries(genres).map(([key, value]) => (
```

Comment:
Same as above - consider going through the code base and make sure those `*Mapper` functions are used similarly, i.e. if you have a map, you then have a separate `*Mapper` function.